src/Authentication/Authentication/test/Find-MgGraphPermission.Tests.ps1

```diff
@@ -75,40 +88,48 @@ Describe "the Find-MgGraphPermission Command" {
             $test = Find-MgGraphPermission 'mail'
             $test | Should -Not -Be $null
             $test.length | Should -Be 17
-            $test.PermissionType[0] | Should -Be 'User'
+            $test.Consent[0] | Should -Be 'User'
             $test.PermissionType[-1] | Should -Be 'Application'
             $test.Id.GetType() | Should -Be 'System.Object[]'
+            $permissions_MsGraphServicePrincipal | Should -Not -Be $null
+            $fromInvokeMgGraphRequest | Should -Be $false
+            { Find-MgGraphPermission 'ReadWrite' -Online | Out-Null } | Should -Throw 'mock connection error message'
         }
 
         It "Returns nothing and throws no exception if a search string is specified and there is no match" {
             { Find-MgGraphPermission 'Nigeria has the best jollof' | Out-Null } | Should -Not -Throw
             Assert-MockCalled Invoke-MgGraphRequest
             $test = Find-MgGraphPermission 'Nigeria has the best jollof'
             $test | Should -Be $null
+            $permissions_MsGraphServicePrincipal | Should -Not -Be $null
+            $fromInvokeMgGraphRequest | Should -Be $false
+            { Find-MgGraphPermission 'Nigeria has the best jollof' -Online | Out-Null } | Should -Throw 'mock connection error message'
         }
     }
 
     Context "When executing the command after initially not having a connection to MS Graph but getting it later" {
         BeforeEach {
-            #$staticVariable = 'No Connection'
-
             Mock Invoke-MgGraphRequest {
-
                 Throw [System.Management.Automation.ValidationMetadataException]::new('mock connection error message')
-                
             }
         }
 
         It 'Executes successfully with no parameters' {
             { Find-MgGraphPermission | Out-Null } | Should -Not -Throw
             Assert-MockCalled Invoke-MgGraphRequest
+            $permissions_MsGraphServicePrincipal | Should -Not -Be $null
+            $fromInvokeMgGraphRequest | Should -Be $false
+            { Find-MgGraphPermission -Online | Out-Null } | Should -Throw 'mock connection error message'
 
             Mock Invoke-MgGraphRequest{
```

Comment:
You might want to add a comment here that this is how you are "restoring" the connection that was set to fail in the `BeforeAll`